ulib/axlibc/c/fnmatch.c

```diff
@@ -9,16 +9,268 @@
 
 #include <fnmatch.h>
 #include <stdio.h>
+#include <string.h>
+#include <ctype.h>
 
 #define END         0
 #define UNMATCHABLE -2
 #define BRACKET     -3
 #define QUESTION    -4
 #define STAR        -5
 
-// TODO
+static int str_next(const char *str, size_t n, size_t *step)
+{
+	if (!n) {
+		*step = 0;
+		return 0;
+	}
```

Comment:
In musl libc, there are more code like `str[0] >= 128U`. Why ignore these codes?  Or add comment `TODO` here.

Reply:
Now that Rukos has not implemented wide characters, these codes can be added after Rukos implements wide characters